0.preprocess-sites/2.process-cells.py

```diff
@@ -0,0 +1,168 @@
+prefilter_file = prefilter_args["prefilter_file"]
+
+foci_dir = spot_args["output_spotdir"]
+
+sort_col = cell_args["sort_col"]
+output_basedir = cell_args["output_basedir"]
+merge_info = cell_args["merge_columns"]
+metadata_merge_left = cell_args["metadata_merge_columns"]["left"]
+metadata_merge_right = cell_args["metadata_merge_columns"]["right"]
+metadata_merge_cells_left = cell_args["metadata_merge_columns"]["cells_left"]
+
+cell_quality = CellQuality(quality_func)
+cell_category_dict = cell_quality.define_cell_quality()
+cell_category_df = pd.DataFrame(cell_category_dict, index=["Cell_Class"])
+
+# Enables feature filtering by loading the Cell Painting feature file.
+# 0.prefilter-features.py must be run first
+try:
+    all_feature_df = pd.read_csv(prefilter_file, sep="\t")
+except FileNotFoundError:
+    raise FileNotFoundError(
+        "Error",
+        f"{prefilter_file} not found.  ",
+        "Perform 0.prefilter-features.py prefilter before continuing..."
+    )
+
+sites = [x for x in os.listdir(batch_dir) if x not in ignore_files]
+
+for site in sites:
+    try:
+        print(f"Now processing {site}...")
+        output_folder = pathlib.Path(output_basedir, batch, "paint", site)
+        os.makedirs(output_folder, exist_ok=True)
+
+        meta_output_file = pathlib.Path(output_folder, f"metadata_{site}.tsv.gz")
+        count_output_file = pathlib.Path(output_folder,f"cell_counts_{site}.tsv")
+
+        compartment_dir = pathlib.Path(batch_dir, site)
+
+        #Make the compartment_csvs dictionary used to merge dfs
+        compartment_csvs = {}
+        for compartment in compartments:
+            try:
+                metadata_cols = parent_cols[compartment.lower()] + id_cols
+            except KeyError:
+                metadata_cols = id_cols
+            compartment_csvs[compartment] = load_single_cell_compartment_csv(
+                compartment_dir, compartment, metadata_cols
+            )
+
+        #Load .tsv created in 1.process-spots
+        foci_file = pathlib.Path(foci_dir, site, "full_cell_category_scores_by_guide.tsv.gz")
+        try:
+            foci_df = pd.read_csv(foci_file, sep="\t")
+        except FileNotFoundError:
+            print("""Run 1.process-spots before continuing. \n
+                  If it has been run, check that output_basedir is set correctly in
+                  the process-spots section of the config""")
+
+        #Relabel columns in foci_df to start with Metadata_Foci_
+        foci_df.columns = ["Metadata_Foci_{}".format(x) for x in foci_df.columns]
+
+    except FileNotFoundError:
+        print(f"{site} data not found")
+        continue
+
+
+    # Merge compartment csvs
+    sc_merged_df = merge_single_cell_compartments(compartment_csvs, merge_info, id_cols)
+    sc_merged_df = sc_merged_df.sort_values(by=sort_col)
+
+    # Add metadata
+    sc_merged_df = sc_merged_df.rename({"Metadata_Cells_ImageNumber": "Metadata_ImageNumber"}, axis="columns")
```

Comment:
This rename step is necessary for the following merge. 
I wasn't sure the right way to extract the hardcoded column names out?

Reply:
I like the strategy in https://github.com/broadinstitute/pooled-cell-painting-profiling-recipe/pull/13/files#diff-d7c74c21a3b3d2924816aa921e827762R80

If we change it to `Metadata_Cells_ImageNumber` then a renaming isn't necessary.